main.tf

```diff
@@ -60,6 +60,7 @@ data "aws_ami" "instance_ami" {
 module "controller_vm" {
   source = "./modules/vms"
 
+  count = var.number_of_controller_instances
   deployment_id = var.deployment_id
   instance_name_suffix = random_string.instance_name_suffix.result
   vm_name_prefix = "controller-"
```

Comment:
for vm_name_prefix change to "controller-${count.index + 1}-"

Reply:
complete